tests/test_composite.py

```diff
@@ -324,18 +324,35 @@ def test_multi_slice_index(ant, sphere, uniform, airplane, globe):
         assert id(sub[i]) == id(multi[i])
         assert sub.get_block_name(i) == multi.get_block_name(i)
     sub = multi[0:-1]
-    assert len(sub) == len(multi) == multi.n_blocks
-    for i in range(multi.n_blocks):
+    assert len(sub) + 1 == len(multi) == multi.n_blocks
+    for i in range(multi.n_blocks - 1):
         assert id(sub[i]) == id(multi[i])
         assert sub.get_block_name(i) == multi.get_block_name(i)
     sub = multi[0:-1:2]
-    assert len(sub) == 3
-    for i in range(3):
+    assert len(sub) == 2
+    for i in range(2):
         j = i*2
         assert id(sub[i]) == id(multi[j])
         assert sub.get_block_name(i) == multi.get_block_name(j)
 
 
+def test_slice_defaults(ant, sphere, uniform, airplane, globe):
+    multi = multi_from_datasets(ant, sphere, uniform, airplane, globe)
+    assert multi[:] == multi[0:len(multi)]
+
+
+def test_slice_negatives(ant, sphere, uniform, airplane, globe):
+    multi = multi_from_datasets(ant, sphere, uniform, airplane, globe)
+    reverse_multi = multi_from_datasets(globe, airplane, uniform, sphere, ant)
+    assert multi[::-1] == reverse_multi
+
+    assert multi[-2:] == multi_from_datasets(airplane, globe)
+
+    assert multi[:-1] == multi_from_datasets(ant, sphere, uniform, airplane)
+
+    assert multi[-1:-4:-2] == multi_from_datasets(globe, uniform)
```

Comment:
These tests would also be a bit more maintainable if you put the meshes in a list and slice them here.

Reply:
I was hoping to avoid these thorny issues and just fix the slicing but I've found that setitem doesn't preserve the name correctly, which makes the testing tricky here.  I need to more fully digest your comments in the issue before fixing more here.  I would prefer having no string key access to the clocks in getitem and setitem.  This would make everything easier.

Reply:
To be clear I just meant changing
```py
multi[-2:] = multi_from_datasets(airplane, globe)
```
to
```py
multi[-2:] = multi_from_datasets(*list_of_meshes[-2:])
```
This should not change the semantics, just make the arguments to `multi_from_datasets` more robust/readable.

Reply:
I was trying to respond to your comment above this one, but I can't seem to directly do so, sorry for the confusion

Reply:
Ah, gotcha, no worries. I don't know why github sometimes does that. You can usually reply from the /files tab of the PR even then.